grype/match/type.go

```diff
@@ -8,12 +8,14 @@ const (
 	ExactDirectMatch   Type = "exact-direct-match"
 	ExactIndirectMatch Type = "exact-indirect-match"
 	CPEMatch           Type = "cpe-match"
+	PURLMatch          Type = "purl-match"
 )
 
 var typeOrder = map[Type]int{
 	ExactDirectMatch:   1,
 	ExactIndirectMatch: 2,
 	CPEMatch:           3,
+	PURLMatch:          4,
```

Comment:
I don't think we want a new match type. This is `exact-direct` since we are searching a vulnerability source we know should govern this package for this exact package. (Indirect is its own type because it signals that we searched via an upstream/source package name, because CPE is its own way of searching. But a PURL search is effectively an exact direct search in this case).